u8timeseries/models/statistics.py

```diff
@@ -21,50 +24,59 @@ def check_seasonality(ts: 'TimeSeries', m: int = None, max_lag: int = 24, alpha:
 
     n_unique = np.unique(ts.values()).shape[0]
 
-    if n_unique > 1:  # Check for non-constant TimeSeries
+    if n_unique == 1:  # Check for non-constant TimeSeries
+        return False, 0
+
+    r = acf(ts.values(), nlags=max_lag)  # In case user wants to check for seasonality higher than 24 steps.
 
-        r = acf(ts.values(), nlags=max_lag)  # In case user wants to check for seasonality higher than 24 steps.
+    gradient = np.gradient(r)
+    gradient_signs_changes = np.diff(np.sign(gradient))
 
-        grad = np.gradient(r)
-        signs_changes = np.diff(np.sign(grad))
+    if m is None:
+        # Tries to infer seasonality from AutoCorrelation Function if no value of m has been provided.
+        # We look for the first positive significant local maximum of the ACF by checking the sign changes
+        # in the gradient.
 
-        if m is None:
-            # Tries to infer seasonality from AutoCorrelation Function if no value of m has been provided.
-            # We look for the first positive local maximum of the ACF by checking the sign changes in the gradient.
+        # Local maximum is indicated by signs_change == -2.
+        if len(np.nonzero((gradient_signs_changes == -2))[0]) == 0:
+            return False, 0
 
-            # Local maximum is indicated by signs_change == -2
-            if len(np.nonzero((signs_changes == -2))[0]) > 0:
-                m = np.nonzero((signs_changes == -2))[0][0] + 1
-            else:
-                return False, 0
+        candidate = np.nonzero((gradient_signs_changes == -2))[0].tolist()
 
-        # Check for local maximum when m is user defined
-        if np.nonzero((signs_changes == -2))[0][0] != m-1:
+    else:
+        # Check for local maximum when m is user defined.
+        if m - 1 not in np.nonzero((gradient_signs_changes == -2))[0]:
             return False, m
 
-        # Remove r[0], the auto-correlation at order 0, that introduces bias.
-        r = r[1:]
+        candidate = [m - 1]
 
-        stat = bartlett_formula(r, m - 1, len(ts))
+    # Remove r[0], the auto-correlation at lag order 0, that introduces bias.
+    r = r[1:]
 
-        # The upper limit of the significance interval.
-        band_upper = r.mean() + norm.ppf(1 - alpha/200)*r.var()
+    # The non-adjusted upper limit of the significance interval.
+    band_upper = r.mean() + norm.ppf(1 - alpha/2)*r.var()
 
-        # Only local maximum with positive values are candidates to be tested against significance level.
-        season = r[m-1] > stat * band_upper
+    # Significance test, stops at first admissible value.
+    is_significant = False
+    i = 0
+    c = 0
 
-        return season, m
+    while not is_significant and i < len(candidate):
+        c = candidate[i] + 1
+        stat = _bartlett_formula(r, c - 1, len(ts))
 
-    return False, 0
+        is_significant = r[c-1] > stat * band_upper
+        i += 1
 
+    return is_significant, c
```

Comment:
👌

Reply:
I will just check if we can avoid this additional `+ 1` and `- 1` later

Reply:
https://github.com/unit8co/u8timeseries/pull/25/files/cecfc132de401009cb6a5520db8c008ef3d939c5#r354174857